tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -69,13 +70,23 @@ def __call__(self, i):
                     }
                 })
             # Enhance access policies with displayName, aadType and principalName
-            i['accessPolicies'] = self.enhance_policies(access_policies)
+            graph_call_succeeded, i['accessPolicies'] = self._enhance_policies(access_policies)
 
         # Ensure each policy is
         #   - User is whitelisted
         #   - Permissions don't exceed allowed permissions
         for p in i['accessPolicies']:
-            if p[self.key] not in self.users:
+            if self.key not in p:
+                # Raise errors when the graph call fails and we need to measure against one of
+                # its provided fields, otherwise return False.
+                if graph_call_succeeded or self.key not in self.GRAPH_PROVIDED_KEYS:
+                    return False
```

Comment:
You still need to check permissions level here (line 90)

Intention of this filter is:
- If user is in the list -- everything is OK
- If user not in the list, but his permissions are less than specified -- everything is OK

If user not in the list AND his permissions are higher than specified -- not OK.

Reply:
Got it, that makes sense.

Reply:
Done.